corehq/apps/app_manager/suite_xml/sections/sync_requests.py

```diff
@@ -0,0 +1,129 @@
+from corehq.apps.app_manager import id_strings
+from corehq.apps.app_manager.suite_xml.contributors import SuiteContributorByModule
+from corehq.apps.app_manager.suite_xml.sections.details import DetailsHelper
+from corehq.apps.app_manager.suite_xml.xml_models import (
+    Command,
+    Display,
+    PushFrame,
+    QueryData,
+    QueryPrompt,
+    SessionDatum,
+    Stack,
+    StackDatum,
+    SyncRequest,
+    SyncRequestPost,
+    SyncRequestQuery,
+    SyncRequestSession,
+    Text,
+)
+from corehq.apps.app_manager.xpath import XPath, CaseTypeXpath
+from corehq.apps.case_search.models import CALCULATED_DATA, MARK_AS_CLAIMED
+from corehq.util.view_utils import absolute_reverse
+
+
+RESULTS_INSTANCE = 'results'  # The name of the instance where search results are stored
+
+
+class SyncRequestContributor(SuiteContributorByModule):
+    """
+    Adds a sync-request node, which sets the URL and query details for
+    synchronous searching and case claiming.
+
+    Search is available from the module's case list.
+
+    See "sync-request" in the `CommCare 2.0 Suite Definition`_ for details.
+
+
+    .. _CommCare 2.0 Suite Definition: https://github.com/dimagi/commcare/wiki/Suite20#sync-request
+
+    """
+    def get_module_contributions(self, module):
+        from corehq.apps.app_manager.models import AdvancedModule, Module
+
+        if isinstance(module, (Module, AdvancedModule)) and module.search_config:
+            domain = self.app.domain
+
+            details_helper = DetailsHelper(self.app)
+
+            sync_request = SyncRequest(
+                post=SyncRequestPost(
+                    url=absolute_reverse('claim_case', args=[domain]),
+                    data=[
+                        QueryData(
+                            key='case_id',
+                            ref=(CaseTypeXpath(module.case_type)
+                                 .case(instance_name=RESULTS_INSTANCE)
+                                 .select(u'@status', u'open', quote=True)
+                                 .select_raw(0)
+                                 .slash(u'@case_id')),
+                            # e.g. instance('results')/results/case[@case_type='foo'][@status='open'][0]/@case_id
```

Comment:
I'm not sure this is right. The `<data>` blocks just add default parameters to the GET request to the URL. We would only want `case_type` here, as this is always required when searching.